templates/default/src/html/partials/base/base-ad-script.hbs

```diff
@@ -1,50 +1,83 @@
 <script>
-	var googletag = googletag || {};
-	googletag.cmd = googletag.cmd || [];
-
-	(function () {
-		var gads = document.createElement("script");
-		gads.async = true;
-		gads.type = "text/javascript";
-		gads.src = "https://www.googletagservices.com/tag/js/gpt.js";
-		var node = document.getElementsByTagName("script")[0];
-		node.parentNode.insertBefore(gads, node);
-	})();
-	var slots = {};
-	googletag.cmd.push(function () {
-		var pgTargeting = {};
-		var section = '{{@root.meta.sectionUrl}}'.replace(/http[s]*?:\/\/(www\.)?/, '');
-
-		googletag.defineSlot('/61381659/' + section, [300, 250], 'ad_bigbox1')
-				.addService(googletag.pubads())
-				.setTargeting('pos', ['btf', 'bigbox1']);
-
-		section.split('/').forEach(function setSValueTargeting(targeting, index) {
-					var key = 's' + (index + 1);
-					pgTargeting[key] = targeting;
-				});
-
-		pgTargeting.pgtype = 'article';
-
-		{{#if @root.meta.keywords}}
-			pgTargeting.meta = '{{@root.meta.keywords}}'.split(',')
-					.map(function removeWhitespaceFromKeywords(keyword) {
-						return keyword.trim();
-					});
-		{{/if}}
-
-		pgTargeting.pageurl = '{{@root.meta.url}}'
-				.replace('https://apps.bostonglobe.com', '')
-				.replace(/\//g, '_');
-
-		for (var k in pgTargeting) {
-			if (pgTargeting.hasOwnProperty(k)) {
-				googletag.pubads().setTargeting(k, pgTargeting[k]);
-			}
-		}
-
-		googletag.pubads().enableAsyncRendering();
-		googletag.pubads().enableSingleRequest();
-		googletag.enableServices();
-	});
+    var googletag = googletag || {};
+    googletag.cmd = googletag.cmd || [];
+
+    (function () {
+        var gads = document.createElement("script");
+        gads.async = true;
+        gads.type = "text/javascript";
+        gads.src = "https://www.googletagservices.com/tag/js/gpt.js";
+        var node = document.getElementsByTagName("script")[0];
+        node.parentNode.insertBefore(gads, node);
+    })();
+
+    var dfpAdUnits = {};
+
+    googletag.cmd.push(function () {
+        var pgTargeting = {};
+        var section = '{{@root.meta.sectionUrl}}'.replace(/http[s]*?:\/\/(www\.)?/, '');
+        var adUnit = '/61381659/' + section;
+
+        {{#each @root.meta.ads}}
```

Comment:
Also, currently `meta.ads` is just a true/false flag. Would the new functionality change that into an array of ad slot strings?

Reply:
Yes, I think it would be best to define that ads that are going to be used in the meta. We need to know which ads are going to be slotted before we define them. meta.ads seemed like the best place in the hierarchy.

Is there other code that needs to be adjusted to make this work?

Reply:
Nope, was just making sure. That makes a lot of sense.

Reply:
Perfect, thanks!